Pay/PaySession.swift

```diff
@@ -277,6 +292,14 @@ extension PaySession: PKPaymentAuthorizationControllerDelegate {
             Log("Checkout doesn't require shipping. Updating address...")
             
             self.delegate?.paySession(self, didUpdateShippingAddress: payPostalAddress, checkout: self.checkout, provide: { updatedCheckout in
+                if let updatedCheckout = updatedCheckout, self.currency.currencyCode != updatedCheckout.currencyCode {
+                    controller.dismiss {
```

Comment:
When solving my ticket: [CON-4037](https://tapcart.atlassian.net/browse/CON-4037) Manish.S/Morgan agreed that is too intrusive to force close the native apple pay dialog on the user automatically. Therefore, in order to solve my ticket for currency mismatch for apple pay on PDP, we decided to wait until user authorizes the payment, to then notify them of the currency mismatch and then restart the apple pay session with the correct currency.

Reply:
thanks @julio-tc – given our conversation earlier, this PR could either be closed or we can rethink how we should evaluate the `PaySession.currencyCode` and `Storefront.Checkout.currency`.

I think the more functionality we can wrap into the SDK PaySession the better. So failing the Authorization from within the PaySession and delegating the message back to the Client can allow for an easier interface in our application. 

So instead of having to track `EdgeCaseLastKnownCheckout`, we call `didReceiveNewCurrencyCode` in `didAuthorize` from the PaySession (instead of `didSelectShippingContact`). 

From there, the delegate may display the error modal, and instantly pass the new currency code to a new pay session.

Reply:
I really like that approach!! at the time I worked on my "apple pay via PDP" ticket I wanted to try to solve this without having to fork the SDK, but if moving forward the idea is to start forking the SDK to own certain aspects of the code then it makes sense to do it this way.